lib/identity/auth.rb

```diff
@@ -226,80 +223,24 @@ class Auth < Sinatra::Base
 
     private
 
-    def call_authorize
-      # if the user is submitting a confirmation form, pull from session,
-      # otherwise get params from the request
-      authorize_params = if params[:authorize]
-        @cookie.authorize_params || {}
-      else
-        filter_params(%w{client_id response_type scope state prompt}).tap do |p|
-          p["scope"] = p["scope"].split(/[, ]+/).sort.uniq if p["scope"]
-        end
-      end
-
-      # clear anything that might be left over in the session
-      @cookie.authorize_params = nil
-
-      begin
-        # Have the user login if:
-        # - We have no session for them
-        # - The client requested that they login
-        if !@cookie.access_token || params[:prompt] == 'login'
-          raise Identity::Errors::LoginRequired
-        end
-
-        # Try to perform an access token refresh if we know it's expired. At
-        # the time of this writing, refresh tokens last 30 days (much longer
-        # than the short-lived 2 hour access tokens).
-        if Time.now > @cookie.access_token_expires_at
-          perform_oauth_refresh_dance
-        end
-
-        # redirects back to the oauth client on success
-        authorize(authorize_params, params[:authorize] == "Allow")
-      # given client_id wasn't found (API throws a 400 status)
-      rescue Excon::Errors::BadRequest
-        flash[:error] = "Unknown OAuth client."
-        redirect to("/login")
-      # we couldn't track the user's session meaning that it's likely been
-      # destroyed or expired, redirect to login
-      rescue Excon::Errors::NotFound
-        redirect to("/login")
-      # user needs to login.
-      rescue Identity::Errors::LoginRequired
+    def redirect_or_authorize
```

Comment:
Assignment Branch Condition size for redirect_or_authorize is too high. [23.35/15]